src/pages/GameMasterView.js

```diff
@@ -116,21 +117,18 @@ const GameMasterView = () => {
                 <AlivePlayersList roomID={roomID} />
                 <DeadPlayersList roomID={roomID} />
             </Flex>
-            <KillButton 
-                arrayOfPlayers={arrayOfPlayers}
-                roomID={roomID}
-                onPlayerKilled={handleKillPlayer}
-                killedPlayerPoints={handleKillPlayerPoints}
-                arrayOfAlivePlayers={arrayOfAlivePlayers}
-            />
-            <AssasinsSelection  
-                roomID={roomID}
-                arrayOfPlayers={arrayOfPlayers} 
-                killedPlayerPoints={killedPlayerPointed}
-                killedPlayerNamed={killedPlayerNamed}
-                triggerAS={triggerAS}
-                setTriggerAS={setTriggerAS} // Pass the setter function as a prop
-            />         
+            <HStack spacing='1px'>
+                <AssasinsSelection
```

Comment:
Let's refactor so that this component and the killbutton are both contained within some parent component which is an intermediate between the game master view and these two components. The intermediate component should contain information like the "assassinPlayerNamed" state variable above. 

This is in line with a broader idea in coding called encapsulation. You can read about it on Wikipedia here but that link is a bit more in a Java context so maybe not the most helpful https://en.wikipedia.org/wiki/Encapsulation_(computer_programming). The idea here, though, is the same as in Java - basically, you want to limit data to who actually needs to know about it. Your game master view shouldn't need to know who is currently selected as the assassin - really it probably needs to know only a small subset of the things the are currently stored as state variables here tbh. Whatever component ends up managing the whole "inputting a kill" mechanism should handle all of the relevant data and logic. That keeps your code easier to read, easier to maintain, and generally more sensible. It makes sense that you should have a component that does "inputting a kill" and that component can be (mostly) put on whatever page you want and still function properly.

There's limits to this of course. When might you want to keep something as a state variable on the game master view page? Mostly if that data is somehow actually integral to the functioning of the page or if many different components on the page all need to know that data and keep it in sync. So stuff like who's alive and who's dead is probably fine as a state variable, since most of our components need to interact with that information regularly.